main.cpp

```diff
@@ -64,14 +64,20 @@ send me a DM to check your pull request
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+
+    int value;
+    std::string name;
+
+    T(int v, const char* s)   //1
+    {
+        value = v; //2
+        name = s; //3
+    }
 };
 
-struct <#structName1#>                                //4
+struct X                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
         if( a->value < b->value ) return a;
         if( a->value > b->value ) return b;
```

Comment:
What must we always do before we use a pointer?
- [ ] revise

Reply:
check for nullptr